### PR TITLE
The compile script support --watch & --packages

### PR DESCRIPTION
**Pull request by imsobear**
- compile support --watch
- compile support -- packages

use for dev:

```bash
node scripts/compile.js --watch --packages rax,rax-server-renderer
```

### CODE REVIEW

**Comment by imsobear** [comment]
@noyobo @yuanyan Please review again 🛩

File: scripts/compile.js [hunk 1]
```diff
@@ -98,14 +112,27 @@ function buildFile(file, silent) {
...
+
+  console.log(chalk.green('watch packages compile', packages));
 
-if (files.length) {
-  files.forEach(file => {
-    buildFile(path.resolve(file), false);
+  chokidar.watch(watchPackagesDir, {
+    ignored: IGNORE_PATTERN
+  }).on('all', (event, filePath) => {
+    if (event !== 'change') {
```
Comment (noyobo) [inline scripts/compile.js]: Also support `add` better?

File: scripts/compile.js [hunk 2]
```diff
@@ -98,14 +112,27 @@ function buildFile(file, silent) {
...
+  const watchPackagesDir = packages.map(dir => path.resolve(dir, 'src'));
+
+  console.log(chalk.green('watch packages compile', packages));
 
-if (files.length) {
-  files.forEach(file => {
-    buildFile(path.resolve(file), false);
+  chokidar.watch(watchPackagesDir, {
+    ignored: IGNORE_PATTERN
+  }).on('all', (event, filePath) => {
```
Comment (noyobo) [inline scripts/compile.js]: If only pass `change`, use `on('change')`.

**Comment by imsobear** [comment]
only support `change`, done!

**Comment by lynnking572** [comment]
--------- Forwarded message ----------
From: "LynnsWorld13" <noreply@github.com>
Date: Jan 20, 2017 12:02 AM
Subject: You've been made an owner of the LynnsWorld13 organization
To: "lynnking572" <Lynnking572@gmail.com>
Cc: 

You have been granted admin access to LynnsWorld13. Admins have full rights to the organization and have complete access to all repositories and teams.

View all LynnsWorld13 admins: https://github.com/orgs/LynnsWorld13/people?query=role%3Aowner

Read more about organization permissions at: https://help.github.com/articles/what-are-the-different-access-perm

**Pull request merged**